### PR TITLE
add functions for facilitate integration of module

### PR DESCRIPTION
**Pull request by antarus**
Added functions for:
- add a maven property
- add dependency to the dependencieManagement section for maven and gradle

Useful for integrating modules

- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/generator-base.js [hunk 1]
```diff
@@ -1185,8 +1185,71 @@ module.exports = class extends PrivateBase {
...
+                    dependency
+                ]
+            }, this);
+        } catch (e) {
+            this.log(e);
+            this.log(`${chalk.yellow('\nUnable to find ') + fullPath + chalk.yellow(' or missing required jhipster-needle. Reference to ')}maven dependency (groupId: ${groupId}, artifactId:${artifactId}, version:${version})${chalk.yellow(' not added.\n')}`);
+            this.debug('Error:', e);
+        }
+    }
+    /**
```
Comment (jdubois) [inline generators/generator-base.js]: Can you add spaces before your functions, for formatting?

**Comment by jdubois** [comment]
LGTM, congrats on your first PR!

**Comment by antarus** [comment]
yes ... you have to start one day: D

File: generators/generator-base.js [hunk 2]
```diff
@@ -1286,6 +1352,34 @@ module.exports = class extends PrivateBase {
...
+    addGradleDependencyManagement(scope, group, name, version) {
+        const fullPath = 'build.gradle';
+        let dependency = `${group}:${name}`;
+        if (version) {
+            dependency += `:${version}`;
+        }
+        try {
+            jhipsterUtils.rewriteFile({
+                file: fullPath,
+                needle: 'jhipster-needle-gradle-dependency-management',
```
Comment (deepu105) [inline generators/generator-base.js]: where is the needle in gradle file for this?

Comment (antarus) [reply generators/generator-base.js]: erf resolve conflict error.
I correct that tonight..

Comment (antarus) [reply generators/generator-base.js]: what is the best way to add unit test on generated code ? sorry node noob :(

Comment (jdubois) [reply generators/generator-base.js]: have a look at the `test` directory, in `generator-jhipster`.
You can also generate tests in the application, so they match the generated code.

**Comment by jdubois** [comment]
This is just what my friend @ludoch needs to finalize the Google App Engine support! So I'm merging this as I know he will need this (he will also be your first tester, hopefully!) - @antarus if you have time to add unit test later, they are of course welcome!!!

And congrats for your first merged PR :-)

**Comment by ludoch** [comment]
Will try soon, thanks!

**Comment by jdubois** [comment]
@antarus can you believe that 3 days after submitting your PR, your code would be used by Google? Now that's the power of Open Source :-)

**Comment by antarus** [comment]
No I did not think lol
For unit tests, I look tomorrow, how it is done under node, and I add them for the generator-jhipster and generator-jhipster-spring-cloud-stream

**Pull request merged**